core/src/main/java/com/datastrato/gravitino/Entity.java

```diff
@@ -41,6 +43,35 @@ public static EntityType fromShortName(String shortName) {
       }
       throw new IllegalArgumentException("Unknown entity type: " + shortName);
     }
+
+    /**
+     * Returns the parent entity types of the given entity type. The parent entity types are the
+     * entity types that are higher in the hierarchy than the given entity type. For example, the
+     * parent entity types of a table are schema, catalog, and metalake.
+     *
+     * @param entityType The entity type for which to get the parent entity types.
+     * @return The parent entity types of the given entity type.
+     */
+    public static List<EntityType> getParentEntityTypes(EntityType entityType) {
+      switch (entityType) {
+        case METALAKE:
+          return ImmutableList.of();
+        case CATALOG:
+          return ImmutableList.of(METALAKE);
+        case SCHEMA:
+          return ImmutableList.of(METALAKE, CATALOG);
+        case TABLE:
+          return ImmutableList.of(METALAKE, CATALOG, SCHEMA);
+        case FILESET:
+          return ImmutableList.of(METALAKE, CATALOG, SCHEMA);
+        case TOPIC:
+          return ImmutableList.of(METALAKE, CATALOG, SCHEMA);
+        case COLUMN:
+          return ImmutableList.of(METALAKE, CATALOG, SCHEMA, TABLE);
+        default:
+          throw new IllegalArgumentException("Unknown entity type: " + entityType);
```

Comment:
Can you please rebase your PR based on @qqqttt123 newly added USER and GROUP support.